src/bindings/python/tests/test_graph/test_custom_op.py

```diff
@@ -197,3 +199,93 @@ def test_custom_op():
     input_tensor = Tensor(input_data)
     results = request.infer({"data": input_tensor})
     assert np.allclose(results[list(results)[0]], expected_output, 1e-4, 1e-4)
+
+
+class CustomSimpleOp(Op):
+    def __init__(self, inputs=None):
+        super().__init__(self)
+        if inputs is not None:
+            self.set_arguments(inputs)
+            self.constructor_validate_and_infer_types()
+
+    def validate_and_infer_types(self):
+        self.set_output_type(0, self.get_input_element_type(0), self.get_input_partial_shape(0))
+
+
+class CustomSimpleOpWithAttribute(Op):
+    class_type_info = DiscreteTypeInfo("CustomSimpleOpWithAttribute", "extension")
+
+    def __init__(self, inputs=None, attrs=None):
+        super().__init__(self)
+        self._attrs = attrs
+        if attrs is not None or inputs is not None:
+            self.set_arguments(inputs)
+            self.constructor_validate_and_infer_types()
+
+    def validate_and_infer_types(self):
+        self.set_output_type(0, self.get_input_element_type(0), self.get_input_partial_shape(0))
+
+    @staticmethod
+    def get_type_info():
+        return CustomSimpleOpWithAttribute.class_type_info
+
+    def visit_attributes(self, visitor):
+        if self._attrs:
+            visitor.on_attributes(self._attrs)
+        return True
+
+
+def test_op_extension(prepared_paths):
+    input_shape = [2, 1]
+
+    core = Core()
+    core.add_extension(CustomSimpleOp)
+    core.add_extension(OpExtension(CustomSimpleOpWithAttribute))
+    core.add_extension(OpExtension(CustomAdd))
+    param1 = ops.parameter(Shape(input_shape), dtype=np.float32, name="data1")
+    param2 = ops.parameter(Shape(input_shape), dtype=np.float32, name="data2")
+    custom_simple = CustomSimpleOp(inputs=[param1, param2])
+    custom_simple.set_friendly_name("test_add")
+    custom_with_attribute = CustomSimpleOpWithAttribute(inputs=[custom_simple], attrs={"value_str": "test_attribute"})
+    custom_add = CustomAdd(inputs=[custom_with_attribute])
+    res = ops.result(custom_add, name="result")
+    simple_model = Model(res, [param1, param2], "SimpleModel")
+
+    cloned_model = simple_model.clone()
+    assert compare_models(simple_model, cloned_model)
+
+    xml_path, bin_path = prepared_paths
+    serialize(simple_model, xml_path, bin_path)
+    model_with_custom_op = core.read_model(xml_path)
+    assert compare_models(simple_model, model_with_custom_op)
+
+
+def test_fail_create_op_ext():
+    class BrokeOp(Op):
+        class_type_info = "BrokeOp"
+
+        def __init__(self, inputs=None):
+            super().__init__(self)
+            if inputs is not None:
+                self.set_arguments(inputs)
+                self.constructor_validate_and_infer_types()
+
+        def get_type_info(self):
+            return BrokeOp.class_type_info
+
+    core = Core()
+    with pytest.raises(RuntimeError) as e:
+        core.add_extension(CustomOp)
```

Comment:
Should it be `BrokeOp` instead of `CustomOp`?

Reply:
I used `CustomOp` from the initial support of custom operations. According to the new rules we should set ctors arguments to None, but this `CustomOp` doesn't meet this requirement. So I decided to reuse it to check it.